pkg/model/bootstrapscript.go

```diff
@@ -245,7 +245,7 @@ func (b *BootstrapScriptBuilder) ResourceNodeUp(c *fi.ModelBuilderContext, ig *k
 
 		// Bastions can have AdditionalUserData, but if there isn't any skip this part
 		if len(ig.Spec.AdditionalUserData) == 0 {
-			return fi.NewStringResource(""), nil
+			return nil, nil
```

Comment:
I see that this avoids returning a resource, and that we wind up setting the "UserData" field of the `awstasks.LaunchTemplate` to nil. How does that change avoid creation of a new launch template. Is it that two different `*fi.StringResource` values don't compare as equal, so _kOps_ thinks that there's a material change required in the launch template?

Reply:
AWS seems to return `nil` when `UserData == ""`. This means that kOps tries to reconcile it.